sd-validation/index.js

```diff
@@ -81,3 +84,53 @@ module.exports = async function validate(textInput) {
 
   return errors;
 };
+
+function getLineNumberFromJsonPath(obj, path) {
+  // To avoid having an extra dependency on a JSON parser we set a unique key in the
+  // object and then use that to identify the correct line
+  const searchKey = Math.random().toString();
+  obj = JSON.parse(JSON.stringify(obj));
+
+  setValueAtJsonLdPath(obj, path, searchKey);
+  const jsonLines = JSON.stringify(obj, null, 2).split('\n');
```

Comment:
The line number we report doesn't apply to the JSON that's passed in, but rather to the stringified JSON. That's not an issue for the feature, but I'm it seems weird when using the sd-validation library as an API consumer.

Should we return the applicable JSON from the validate function?

Reply:
I agree it's weird from the library perspective, but I think we can punt and just file an issue/TODO/jsdoc comment to mark this explicitly.